nomic/dataset.py

```diff
@@ -1097,10 +1098,44 @@ def create_index(
 
         self._latest_dataset_state()
 
-        if isinstance(projection, Dict):
-            projection = NomicProjectOptions(**projection)
-        else:
-            projection = NomicProjectOptions()
+        projection_algorithm = "nomic-project"
+        nomic_projection = None
+        umap_projection = None
+
+        if isinstance(projection, UMAPOptions):
+            projection_algorithm = "umap"
+            umap_projection = projection
+        elif isinstance(projection, NomicProjectOptions):
+            nomic_projection = projection
+        elif isinstance(projection, Dict):
+            if projection.get("algorithm", "").lower() == "umap":
+                projection_algorithm = "umap"
+                try:
+                    umap_projection = UMAPOptions(**{k: v for k, v in projection.items() if k != "algorithm"})
+                except Exception as e:
+                    raise ValueError(f"Could not instantiate UMAPOptions from dict: {e}")
+            else:
+                nomic_projection = NomicProjectOptions(**projection)
+        elif projection is True:
+            nomic_projection = NomicProjectOptions()
```

Comment:
we should send something like generic projection option instead of setting umap or nomic project